### ISSUE TITLE
RFC: Popup more intuitive behavior with multiple `on` events

### ISSUE THREAD

**Issue by mauretanec** [Semantic-Org/Semantic-UI-React#2765]
## Bug Report
The problem happens when using multiple events to trigger Popup.
Can be reproduced with `<Popup trigger={<button>Show tooltip</button>} content='tooltip' on={['click', 'hover']}` but the easiest way is with `<Popup trigger={<button>Show tooltip</button>} content='tooltip' on={['click', 'focus']}`.
Just try clicking the button. For the first time, it opens the popup and immediately closes it. Following clicks work correctly.
This problem has been mentioned in #2075 by @austinfox [here](https://github.com/Semantic-Org/Semantic-UI-React/issues/2075#issuecomment-331204037).

### Steps
1. Go to [Semantic UI React Popup demo](https://react.semantic-ui.com/modules/popup#popup-example).
2. Edit e.g. first example `Popup` component to contain multiple on events i.e. `on={['click', 'focus']}`.
3. Click the popup trigger.

### Expected Result
Popup is displayed and stays open.

### Actual Result
Popup is displayed and closed immediately.

### Version
0.80.0

### Testcase
Check the steps section above.

**Comment by levithomason** [comment]
I'm not sure this is a bug necessarily.  The order of events in the browser when clicking a button is `click` then `focus`.  What is happing is technically correct, though not helpful.  The first event, click, opens the popup while the immediate second event, focus, closes it again.

The `on` prop _toggles_ the popup on the given events.  There are a few ways this could be reworked to give more intuitive results.

### openOn / closeOn

Splitting or adding these props would allow you to specify which events should open and close the popup.  You'd apply focus to the `openOn` and click to both `openOn` and `closeOn`.  This way, focus only opens the popup.

### Special focus/blur handling

This might be the more desirable solution.  Focus could strictly open the popup and blur could strictly close it.  This is not technically the correct behavior given the prop config of "toggle on focus", however, it does give a more intuitive behavior.  I would think most users would expect focus to open the popup.

I'd like to hear thoughts and feedback on these and any other solutions folks may have.

**Comment by mauretanec** [comment]
@levithomason: I agree with latter proposal (special focus/blur handling) as it complies with accessibility guidelines - receiving focus should show the popup or tooltip and blur should hide it.

**Comment by levithomason** [comment]
Agreed! A PR implementing open on focus and close on blur is welcome.
